uk.co.compucorp.civicrm.hrcore/tests/phpunit/CRM/HRCore/CMSData/PathsFactoryTest.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
+
+use CRM_HRCore_CMSData_PathsFactory as CMSPathsFactory;
+
+/**
+ * Class CRM_HRCore_CMSData_PathsFactoryTest
+ *
+ * @group headless
+ */
+class CRM_HRCore_CMSData_PathsFactoryTest extends \PHPUnit_Framework_TestCase implements HeadlessInterface, TransactionalInterface {
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->installMe(__DIR__)
+      ->apply();
+  }
+
+  public function testItReturnsAClassWithImplementingTheExpectedInterface() {
+    $contactData = [];
+    $pathsClass = CMSPathsFactory::create('Drupal', $contactData);
+
+    $this->assertTrue($pathsClass instanceof CRM_HRCore_CMSData_PathsInterface);
+  }
+
+  public function testItThrowsAnExceptionIfGivenAnUnrecognizedCMSName() {
+    $cmsName = 'UnrecognizedCMS';
+
+    $this->setExpectedException('Exception', "CMS \"{$cmsName}\" not recognized");
```

Comment:
this method is  removed in newer phpunit versions, you need to use annotation instead : https://phpunit.de/manual/current/en/appendixes.annotations.html#appendixes.annotations.expectedException

Reply:
It wasn't removed it was just renamed to `expectedException()`: https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.exceptions

But yeah, I this specific case, an annotation would work better. Usually, we use `setExpectedException` in there is more than one place in the code that can throw the exception and we want to make sure the test will catch the right one